tests/kafkatest/tests/replication_test.py

```diff
@@ -15,45 +15,86 @@
 
 from ducktape.tests.test import Test
 from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
+from ducktape.mark import parametrize
 
 from kafkatest.services.zookeeper import ZookeeperService
 from kafkatest.services.kafka import KafkaService
 from kafkatest.services.verifiable_producer import VerifiableProducer
 from kafkatest.services.console_consumer import ConsoleConsumer
 
-import signal
 import time
 
 
+"""
+3 brokers, 3 topics, 3 partitions, 3 replicas on each topic
+min.insync.replicas == 2
+ack -1
+failures: [leader, follower, controller] X [clean, hard, soft]
+One test cases with with ack 1
+One test case with compression toggled on
+
+"""
+
+# Failure types
```

Comment:
Seems like we could just split these into two variables:

1. Type of death (clean, hard, soft)
2. Permanence (bounce vs kill).

This would clean up some of the if statements in the code farther down in this file.